### PR TITLE
ecmo-subject.toml

### PR DESCRIPTION
**Pull request by ekamau**
First draft of subjects table for the ECMO dataset. Have a look! Hopefully I have created the branch correctly ... via web interface.

### CODE REVIEW

File: isaric/parsers/ecmo_subject.toml [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+  field = "apsc_gestout"
+  description = "Gestational outcome"
+  values = { 1 = "Term birth", 2 = "Preterm birth" }
+
+[subject.pregnancy_whether_breastfed] # SK to check here!!
+  combinedType = "any"
+  description = "Breastfed"
+  fields = [
+    { field = "apsc_brfedind", ref = "Y/N/NK" },
+    { field = "apsc_brfed", values = { 1 = "YES - currently breastfeeding", 2 = "YES - breastfeeding discontinued", 0 = "NO" } }
```
Comment (abhidg) [inline isaric/parsers/ecmo_subject.toml:95]: field is boolean, so values should map to true/false `values = { 1 = true, 2 = true, 0 = false }`

File: isaric/parsers/ecmo_subject.toml [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+  combinedType = "firstNonNull"
+  fields = [
+    { field = "f_datedeath", description = "Date of death", if = { dsterm = 4 } },
+    { field = "dsstdtc", description = "Outcome date", if = { dsstdtcyn = 1 } }
+  ]
+  
+[subject.icu_admitted]
+  field = "icu_hoterm"	
+  description = "Admitted to ICU?"
+  apply = { function = "isNotNull" } # WHAT DOES THIS DO?
```
Comment (abhidg) [inline isaric/parsers/ecmo_subject.toml:229]: This uses the [isNotNull function](https://github.com/globaldothealth/adtl/blob/9904403d94edb8284016a7c4fbd3189f1dd66c28/adtl/transformations.py#L7) defined in adtl.transformations, so icu_admitted is true if icu_hoterm is not null.

**Review by abhidg — COMMENTED** [review]
Looks good, will try validating. We should decide on a policy of firstNonNull / any for boolean fields, will create a separate ticket for that.

File: isaric/parsers/ecmo_subject.toml (hunk 1 above)
Comment (ekamau) [reply isaric/parsers/ecmo_subject.toml:95]: Thanks. I will correct this.

File: isaric/parsers/ecmo_subject.toml (hunk 2 above)
Comment (ekamau) [reply isaric/parsers/ecmo_subject.toml:229]: Understood.

**Review by pipliggins — APPROVED** [review]

**Review by ekamau — COMMENTED** [review]
Added extra edits

**Comment by ekamau** [comment]
@abhidg - check the new edits and changes regarding admission dates ..

**Comment by pipliggins** [comment]
Validated the parser and added in the visit table from Rosario - the majority of the data appear to be missing an outcome date which Is causing a low valid % for the visits

**Pull request merged**